### ISSUE TITLE
Need to login to EPIC multiple times a week

### ISSUE THREAD

**Issue by Zorrototo** [Heroic-Games-Launcher/HeroicGamesLauncher#2536]
### Describe the bug

Almost every time I try to launch a game, it tells me my credentials are expired, so I log back in with alternative method (tried the normal login too, didn't help), and it woks, until one or two days later, where it tells me my credentials are expired, so rinse and repeat..

I tried to completely wipe the .config/heroic folder and start over, this issue comes back.

### Add logs

```text
(16:16:25) INFO:    [Connection]:       Connectivity: check-online
(16:16:25) INFO:    [Connection]:       Pinging external endpoints
(16:16:25) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(16:16:25) INFO:    [Legendary]:        Legendary location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(16:16:25) INFO:    [Gog]:              GOGDL location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(16:16:25) INFO:    [Gog]:              GOGDL location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(16:16:26) WARNING: [Backend]:          Protocol already registered.
(16:16:26) INFO:    [Gog]:              Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --version
(16:16:26) INFO:    [Connection]:       Connectivity: online
(16:16:26) INFO:    [Backend]:          Loading Screen Ready
(16:16:26) INFO:    [Frontend]:         Refreshing Library
(16:16:26) INFO:    [Legendary]:        Refreshing library...
(16:16:26) INFO:    [Legendary]:        Refreshing Epic Games...
(16:16:26) INFO:    [Legendary]:        Game list updated, got 273 games & DLCs
(16:16:26) INFO:    [Backend]:          Frontend Ready
(16:16:26) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(16:16:26) INFO:    [Backend]:          Checking for new Heroic Updates
(16:16:26) INFO:    [Backend]:          Checking for current version changelog
(16:16:26) INFO:    [Backend]:          

Heroic Version: 2.6.2 Trafalgar Law
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.6

Electron Version: 22.2.0
Chrome Version: 108.0.5359.215
NodeJS Version: 16.17.1

OS: ManjaroLinux KERNEL: 6.1.16-1-MANJARO ARCH: x64
CPU: AMD Ryzen 5 3600 6-Core Processor @3.6 GOVERNOR: schedutil
RAM: Total: 15.55 GiB Available: 13 GiB
GRAPHICS: GPU0: NVIDIA GeForce GTX 1060 6GB VRAM: 6070MB DRIVER: 525.89.02 
PROTOCOL: x11

(16:16:27) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(16:16:27) INFO:    [Legendary]:        Checking for game updates: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(16:16:28) INFO:    [Winetricks]:       Downloading Winetricks
(16:16:30) INFO:    [Legendary]:        Found 0 games to update
(16:16:30) INFO:    [Gog]:              Found 0 game(s) to update
(16:16:31) INFO:    [Backend]:          Starting the Download Queue
(16:16:31) INFO:    [Backend]:          Launching Shadow of the Tomb Raider: Definitive Edition (890d9cf396d04922a1559333df419fed)
(16:16:31) INFO:    [Backend]:          Preventing display from sleep
(16:16:31) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(16:16:32) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(16:16:32) WARNING: [Backend]:          You are using Proton, this can lead to some bugs. Please do not open issues with bugs related to games
(16:16:32) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(16:16:32) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(16:16:32) DEBUG:   [Backend]:          Running Wine command: run wineboot --init
(16:16:33) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 890d9cf396d04922a1559333df419fed --json --offline
(16:16:34) INFO:    [Legendary]:        Launching Shadow of the Tomb Raider: Definitive Edition: LD_PRELOAD= STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/omano/.steam/steam STEAM_COMPAT_DATA_PATH=/home/omano/Games/Heroic/Prefixes/ShadowoftheTombRaider PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 STEAM_COMPAT_APP_ID=0 SteamAppId=0 SteamGameId=heroic-ShadowoftheTombRaider PROTON_LOG_DIR=/home/omano /usr/bin/gamemoderun /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 890d9cf396d04922a1559333df419fed --language en --no-wine --wrapper "'/home/omano/.config/heroic/tools/proton/Proton-GE-Proton7-49/proton' run"
(16:16:34) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(16:16:35) INFO:    [Frontend]:         Refreshing Library
(16:16:35) INFO:    [Legendary]:        Refreshing library...
```

### Steps to reproduce

1. Try to start a game
2. See error

### Expected behavior

You stay logged in the maximum time possible, not only one or two days.

### Heroic Version

Latest Stable

### System Information

- OS : Manjaro KDE

**Comment by Zorrototo** [comment]
I can confirm the same behavior on a friend's computer. He bought Dead By Daylight on EPIC Store, then started Heroic Launcher, the game wasn't in his Library. I force library refresh but it seemed to do the "refresh animation" instantly. I said to him "100% sure this is the bug I reported today, you're not logged in anymore". I went to the Accounts/Login page, I logged him out, I logged him in with alternative login method, I clicked refresh button in his Library, and Dead by Daylight appeared after 2 seconds "refresh animation".

**Comment by matteobtt** [comment]
**Found the problem**. The command `legendary list --third-party`, before getting the list of games, checks if there is a still valid authentication session in the `user.json` file, if it has expired, a call is made to get a new access token via an OAuth refresh token, this call also changes the refresh token invalidating the previous one.

When Heroic starts up each of these two lines:
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/690f522155940d2fc6dff3339a53f38002ee37d9/src/frontend/state/GlobalState.tsx#L469-L470
end up executing the same command without waiting for the first execution to finish before executing the next one, so the second execution fails to renew the authentication because both executions are using the same refresh token, after it fails, the `User.json` file is deleted and when starting a game it shows the error "Your credentials have expired, please log out and log in again.".

### Legendary log
```
[13/3/2023, 21:44:21] /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
legendary version "0.20.32", codename "Dark Energy (hotfix #6)"
[13/3/2023, 21:44:22] /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
[13/3/2023, 21:44:22] /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[cli] INFO: Getting game list... (this may take a while)
[EPCAPI] ERROR: Login to EGS API failed with errorCode: errors.com.epicgames.account.auth_token.invalid_refresh_token
[Core] ERROR: Stored credentials are no longer valid! Please login again.
[cli] ERROR: Login failed, cannot continue!

Available games:
...
```

### Steps to reproduce
1. Open the file `~/.config/legendary/user.json` (or `~/.var/app/com.heroicgameslauncher.hgl/config/legendary/user.json` if you have Heroic installed as Flatpak)
2. Change the field `"expires_at"` to a past date.
3. Start Heroic and then a game.

**Comment by arielj** [comment]
interesting! thanks for the debug and the details, I'll try to reproduce the problem

what's weird to me at first glance is that you say the second line is executed before the first one ends, but that's the `await` for at the beginning, to wait until it ends to then run the next one

**Comment by ax-ross** [comment]
I have same issue too.

**Comment by arielj** [comment]
> ### Steps to reproduce
> 1. Open the file `~/.config/legendary/user.json` (or `~/.var/app/com.heroicgameslauncher.hgl/config/legendary/user.json` if you have Heroic installed as Flatpak)
> 2. Change the field `"expires_at"` to a past date.
> 3. Start Heroic and then a game.

@matteobtt is there anything else that might me missing from these steps? I do that (I replace the expires_at with a past date) and after starting heroic I don't have any login issue, maybe there's something else that is relevant?

I do see 2 calls to the `list` command, but both are succeeding

**Comment by matteobtt** [comment]
@arielj Indeed, the problem is not always reproducible, I have done various tries and seems to be time sensitive, if I try to run the program twice in parallel at the same time, the second run succeed but if I introduce a slight delay of at least 50ms in the second run, it fails, if I increase the delay to 600ms, it succeeds again.

These are my attempts, I added a log line with the last characters of the refresh token, note that in the last attempt the refresh tokens are different:
```
[matteo@matteo-linux legendary]$ legendary list --third-party & legendary list --third-party
[1] 11038
[cli] INFO: Logging in...
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[EPCAPI] INFO: Refresh Token: ...NSloe3F4XK
[Core] INFO: Logging in...
[EPCAPI] INFO: Refresh Token: ...NSloe3F4XK
[EPCAPI] INFO: Status code: 200
[EPCAPI] INFO: Status code: 200
[cli] INFO: Getting game list... (this may take a while)
[cli] INFO: Getting game list... (this may take a while)
```
```
[matteo@matteo-linux legendary]$ legendary list --third-party & sleep 0.05 && legendary list --third-party
[1]+  Completato              legendary list --third-party
[1] 11350
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[EPCAPI] INFO: Refresh Token: ..._21LviDvgD
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[EPCAPI] INFO: Refresh Token: ..._21LviDvgD
[EPCAPI] INFO: Status code: 200
[cli] INFO: Getting game list... (this may take a while)
[EPCAPI] INFO: Status code: 400
[EPCAPI] ERROR: Login to EGS API failed with errorCode: errors.com.epicgames.account.auth_token.invalid_refresh_token
[Core] ERROR: Stored credentials are no longer valid! Please login again.
[cli] ERROR: Login failed, cannot continue!
```
```
[matteo@matteo-linux legendary]$ legendary list --third-party & sleep 0.5 && legendary list --third-party
[1] 12056
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[EPCAPI] INFO: Refresh Token: ...mwZKTDLn7Z
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[EPCAPI] INFO: Refresh Token: ...mwZKTDLn7Z
[EPCAPI] INFO: Status code: 200
[cli] INFO: Getting game list... (this may take a while)
[EPCAPI] INFO: Status code: 400
[EPCAPI] ERROR: Login to EGS API failed with errorCode: errors.com.epicgames.account.auth_token.invalid_refresh_token
[Core] ERROR: Stored credentials are no longer valid! Please login again.
[cli] ERROR: Login failed, cannot continue!
```
```
[matteo@matteo-linux legendary]$ legendary list --third-party & sleep 0.6 && legendary list --third-party
[1] 12013
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[EPCAPI] INFO: Refresh Token: ...GUUxNXR55V
[EPCAPI] INFO: Status code: 200
[cli] INFO: Getting game list... (this may take a while)
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[EPCAPI] INFO: Refresh Token: ...sH7-Q28B3N
[EPCAPI] INFO: Status code: 200
[cli] INFO: Getting game list... (this may take a while)
```

**Comment by ShaneTRS** [comment]
I can confirm this behavior as well. I have had to log into Epic Games's service 3 times today when launching through Heroic. I haven't tested `legendary` stand-alone, but my browser session is never logged out. I also didn't try modifying the `expires_at` value to see if it was Heroic tossing valid credentials, so this could be on Epic's end.

**Comment by CoolPenguin27** [comment]
I can confirm that this behavior happens to me, I generally need to log in daily due to this bug (Ubuntu 22.10).

**Comment by LP5** [comment]
Same on Arch Linux (lts core), Heroic 2.6.2.

**Comment by matteobtt** [comment]
I've noticed that the latest release fixed the problem, can this issue be closed?

**Comment by Zorrototo** [comment]
I mean it just released maybe wait a week before making sure it is fixed.

**Comment by BlueBoxWare** [comment]
Same. Haven't had to re-login since the update 2 weeks ago.

**Comment by matteobtt** [comment]
Unfortunately, the same problem reappeared in version 2.7.1

Heroic Error:
```
An error has occurred! Try to Logout and Login on your Epic account.
gamemodeauto:gamemodeauto:[cli] INFO: Logging in...
Traceback (most recent call last):
File "legendary/cli.py", line 3061, in <module>File "legendary/cli.py", line 2976, in main
File "legendary/cli.py", line 592, in launch_game
File "legendary/core.py", line 187, in login
ValueError: No saved credentials
[257] Failed to execute script 'cli' due to unhandled exception!gamemodeauto:gamemodeauto:
```

Legendary log:
```
[7/5/2023, 20:54:46] /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
legendary version "0.20.32", codename "Dark Energy (hotfix #6)"
[7/5/2023, 20:54:47] /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
[cli] INFO: Logging in...
[7/5/2023, 20:54:48] /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[cli] INFO: Getting game list... (this may take a while)
[Core] INFO: Logging in...
[EPCAPI] ERROR: Login to EGS API failed with errorCode: errors.com.epicgames.account.auth_token.invalid_refresh_token
[Core] ERROR: Stored credentials are no longer valid! Please login again.
[cli] ERROR: Login failed, cannot continue!
```

**Comment by Zorrototo** [comment]
```
(21:41:47) INFO:    [Legendary]:        Refreshing library...
(21:41:47) INFO:    [Legendary]:        Refreshing Epic Games...
(21:41:47) INFO:    [Connection]:       Connectivity: check-online
(21:41:47) INFO:    [Connection]:       Pinging external endpoints
(21:41:47) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(21:41:47) INFO:    [Legendary]:        Legendary location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(21:41:47) INFO:    [Gog]:              GOGDL location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(21:41:47) INFO:    [Gog]:              GOGDL location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(21:41:47) INFO:    [Legendary]:        Game list updated, got 285 games & DLCs
(21:41:47) WARNING: [Backend]:          Protocol already registered.
(21:41:47) INFO:    [Gog]:              Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/omano/.config/heroic/gog_store/auth.json --version
(21:41:47) INFO:    [Connection]:       Connectivity: online
(21:41:47) INFO:    [Backend]:          Loading Screen Ready
(21:41:47) INFO:    [Frontend]:         Refreshing Library
(21:41:47) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(21:41:47) INFO:    [Gog]:              Found 0 game(s) to update
(21:41:47) INFO:    [Backend]:          Frontend Ready
(21:41:47) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(21:41:47) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(21:41:48) INFO:    [Backend]:          

Heroic Version: 2.7.1 Eustass Kid
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.7.1

Electron Version: 23.1.4
Chrome Version: 110.0.5481.192
NodeJS Version: 18.12.1

OS: ManjaroLinux KERNEL: 6.1.26-1-MANJARO ARCH: x64
CPU: AMD Ryzen 5 3600 6-Core Processor @3.6 GOVERNOR: schedutil
RAM: Total: 15.54 GiB Available: 10.61 GiB
GRAPHICS: GPU0: NVIDIA GeForce GTX 1060 6GB VRAM: 6070MB DRIVER: 530.41.03
PROTOCOL: x11

(21:41:48) INFO:    [Legendary]:        Checking for game updates: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(21:41:49) DEBUG:   [Legendary]:        Update is available for 2e26af55c1a146e7a939dcc1014fc9fd: 1.9227.4115.502609227 != 1.10126.4742.502610126
(21:41:49) INFO:    [Legendary]:        Found 1 game to update
(21:41:49) INFO:    [Winetricks]:       Downloading Winetricks
(21:41:52) INFO:    [Backend]:          Starting the Download Queue
(21:41:57) INFO:    [Backend]:          Launching Dead by Daylight (Brill)
(21:41:57) INFO:    [Backend]:          Preventing display from sleep
(21:41:57) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(21:41:57) INFO:    [Backend]:          Preventing machine to sleep
(21:41:57) INFO:    [Backend]:          Stopping Power Saver Blocker
(21:41:57) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(21:41:57) WARNING: [Backend]:          You are using Proton, this can lead to some bugs. Please do not open issues with bugs related to games
(21:41:57) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(21:41:57) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(21:41:57) DEBUG:   [Backend]:          Running Wine command: run wineboot --init
(21:41:59) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch Brill --json --offline
(21:41:59) INFO:    [Legendary]:        Launching Dead by Daylight: LD_PRELOAD= STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/omano/.steam/steam STEAM_COMPAT_DATA_PATH="/home/omano/Games/Heroic/Prefixes/Dead by Daylight" DXVK_FRAME_RATE=75 PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 PROTON_EAC_RUNTIME=/home/omano/.config/heroic/tools/runtimes/eac_runtime STEAM_COMPAT_APP_ID=0 SteamAppId=0 SteamGameId=heroic-DeadByDaylight PROTON_LOG_DIR=/home/omano /usr/bin/gamemoderun /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch Brill --language en --no-wine --wrapper "'/home/omano/.config/heroic/tools/proton/Proton-GE-Proton7-49/proton' run"
(21:42:00) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(21:42:00) INFO:    [Frontend]:         Refreshing Library
(21:42:00) INFO:    [Legendary]:        Refreshing library...
(21:42:19) INFO:    [Legendary]:        Logging out: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary auth --delete
(21:42:19) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary cleanup
(21:42:35) INFO:    [Legendary]:        Logging in: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary auth --code <redacted>
(21:42:36) INFO:    [Frontend]:         Refreshing Library
(21:42:36) INFO:    [Legendary]:        Refreshing library...
(21:42:36) INFO:    [Legendary]:        Refreshing Epic Games...
(21:42:36) ERROR:   [Legendary]:        Failed to parse MtMassive.json
(21:42:36) ERROR:   [Legendary]:        Failed to parse bobcat.json
(21:42:36) INFO:    [Legendary]:        Game list updated, got 285 games & DLCs
(21:42:36) INFO:    [Frontend]:         Called Login
(21:42:36) INFO:    [Frontend]:         No cache found, getting data from legendary...
(21:42:36) INFO:    [Legendary]:        Refreshing library...
(21:42:36) INFO:    [Legendary]:        Refreshing Epic Games...
(21:42:36) ERROR:   [Legendary]:        Failed to parse MtMassive.json
(21:42:36) ERROR:   [Legendary]:        Failed to parse bobcat.json
(21:42:36) INFO:    [Legendary]:        Game list updated, got 285 games & DLCs
(21:42:36) INFO:    [Frontend]:         Force Update
(21:42:37) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(21:42:37) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(21:42:40) INFO:    [Frontend]:         Refreshing Library
(21:42:40) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(21:42:40) INFO:    [Legendary]:        Refreshing library...
(21:42:40) INFO:    [Legendary]:        Refreshing Epic Games...
(21:42:40) INFO:    [Legendary]:        Game list updated, got 285 games & DLCs
(21:42:41) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(21:42:44) INFO:    [Backend]:          Launching Dead by Daylight (Brill)
(21:42:44) INFO:    [Backend]:          Preventing display from sleep
(21:42:44) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(21:42:44) INFO:    [Backend]:          Stopping Power Saver Blocker
(21:42:45) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(21:42:45) WARNING: [Backend]:          You are using Proton, this can lead to some bugs. Please do not open issues with bugs related to games
(21:42:45) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(21:42:45) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-Proton7-49
(21:42:45) DEBUG:   [Backend]:          Running Wine command: run wineboot --init
(21:42:46) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch Brill --json --offline
(21:42:47) INFO:    [Legendary]:        Launching Dead by Daylight: LD_PRELOAD= STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/omano/.steam/steam STEAM_COMPAT_DATA_PATH="/home/omano/Games/Heroic/Prefixes/Dead by Daylight" DXVK_FRAME_RATE=75 PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 PROTON_EAC_RUNTIME=/home/omano/.config/heroic/tools/runtimes/eac_runtime STEAM_COMPAT_APP_ID=0 SteamAppId=0 SteamGameId=heroic-DeadByDaylight PROTON_LOG_DIR=/home/omano /usr/bin/gamemoderun /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch Brill --language en --no-wine --wrapper "'/home/omano/.config/heroic/tools/proton/Proton-GE-Proton7-49/proton' run"

```

**Comment by Zorrototo** [comment]
Here is the popup I have when I start Heroic, and try to launch Dead by Daylight

```
Legendary (Epic Games)
Error:
An error has occurred! Try to Logout and Login on your Epic account.
gamemodeauto:[cli] INFO: Logging in...
Traceback (most recent call last):
File "legendary/cli.py", line 3061, in <module>
File "legendary/cli.py", line 2976, in main
File "legendary/cli.py", line 592, in launch_gameFile "legendary/core.py", line 187, in login
ValueError: No saved credentials
[28046] Failed to execute script 'cli' due to unhandled exception!
```

If I log out and log back in, it then launch the game.

**Comment by Fitzelchen01** [comment]
> Here is the popup I have when I start Heroic, and try to launch Dead by Daylight
> 
> ```
> Legendary (Epic Games)
> Error:
> An error has occurred! Try to Logout and Login on your Epic account.
> gamemodeauto:[cli] INFO: Logging in...
> Traceback (most recent call last):
> File "legendary/cli.py", line 3061, in <module>
> File "legendary/cli.py", line 2976, in main
> File "legendary/cli.py", line 592, in launch_gameFile "legendary/core.py", line 187, in login
> ValueError: No saved credentials
> [28046] Failed to execute script 'cli' due to unhandled exception!
> ```
> 
> If I log out and log back in, it then launch the game.

I can confirm. Same thing here with same error message.
It happens around every second day.

**Comment by CoolPenguin27** [comment]
Can confirm the issue reappeared:

`An error has occurred! Try to Logout and Login on your Epic account.
Traceback (most recent call last):
File "legendary/cli.py", line 3061, in <module>
File "legendary/cli.py", line 3000, in main
File "legendary/cli.py", line 1587, in info
File "legendary/api/egs.py", line 190, in get_user_entitlementsAttributeError: 'NoneType' object has no attribute 'get'
[214] Failed to execute script 'cli' due to unhandled exception!`

**Comment by jony10** [comment]
Same on Windows 11. Happens daily since updating to 2.7.1

[cli] INFO: Logging in...
[Core] INFO: Logging in...
[cli] INFO: Getting game list... (this may take a while)
[EPCAPI] ERROR: Login to EGS API failed with errorCode: errors.com.epicgames.account.auth_token.invalid_refresh_token
[Core] ERROR: Stored credentials are no longer valid! Please login again.
[cli] ERROR: Login failed, cannot continue!
[Core] INFO: Fetching metadata for 1 app(s).

**Comment by jabuxas** [comment]
is it possible to bisect the commit that fixed it and see what broke it again? i've had this problem for at least 6 months now

**Comment by arielj** [comment]
if anyone wants to try the build from this PR for a while https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/2770 it should fix this issue

**Comment by BlueBoxWare** [comment]
> if anyone wants to try the build from this PR for a while #2770 it should fix this issue

So far, between a few reboots, I have not needed to relogin with that build. 

However I encounter another issue with that build: Every time when starting the first game after starting Heroic I get a popup that the game is not installed (even though it is) . After that first failed try, I can start the game or any other installed game without issue.

Sample log (of a failed first start): https://gist.github.com/BlueBoxWare/71f824e2e5995af1fbc7ec966784509a

BTW, if anybody else reading this wants to try this build, you can download the AppImage here: https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/actions/runs/5164789203

**Comment by arielj** [comment]
> However I encounter another issue with that build: Every time when starting the first game after starting Heroic I get a popup that the game is not installed (even though it is) . After that first failed try, I can start the game or any other installed game without issue.
> 
> Sample log (of a failed first start): https://gist.github.com/BlueBoxWare/71f824e2e5995af1fbc7ec966784509a

Weird, the change it that build is minimal and totally unrelated to that issue you are having :S. Maybe the issue is not the change but it being an AppImage executable.

**Comment by BlueBoxWare** [comment]
> Weird, the change it that build is minimal and totally unrelated to that issue you are having :S. Maybe the issue is not the change but it being an AppImage executable.

I tried the the regular AppImage, but the issue doesn't happen there.

**Comment by quartzar** [comment]
Is there progress with this issue? I have to log back in every single reboot.

**Comment by shessafridi** [comment]
Same issue I have to login again after 2 days, I am on Windows 11

**Comment by takesnodamage** [comment]
Just hit what sounds like this issue on 2.8.0. Got GRIME from the Epic store but didn't see it in Heroic, refresh library button completed instantly. Logged out and back in, now GRIME is in the library.

**Comment by Zorrototo** [comment]
To be honest, and this is not a silly way to "pressure" Heroic team, but I don't really use Heroic anymore, since this issue is not resolved, the need to login every single time I start Heroic is a real deterrent for me, and probably to a lot of people, this should be a high priority issue.

**Comment by arielj** [comment]
The fix has been available for some time already in the development branch, I was not released yet (there were problems with other areas in Heroic I think when we tried to release 2.9), but it's possible to use the development builds until it's released:

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/188464/d7b96391-b379-4056-8e26-63f582c411b0)
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/188464/86875400-049f-478e-9c86-bebe1673adb4)

**Comment by Zorrototo** [comment]
All right. Thanks for bringing full info on procedure. But I mean most people will not be able to use a dev branch, this is what I meant when I talked about most people are probably avoiding Heroic currently, and why the fix should be a priority to be pushed to main branch so common installation methods for the mass will have the fix.

**Comment by arielj** [comment]
This should be fixed since 2.9.1. Closing this and can be reopened if it's still happening.

**Comment by zzFluke** [comment]
2.9.1 on Steam Deck still experience required login once a day.

**Comment by bemymonkey** [comment]
Just ran into the issue today. Running 2.9.1 on Debian Sid.

**Comment by LazyDevv** [comment]
Been happening for a few days in a row. V2.9.2 - Windows 10 machine.

**Comment by reicadencell-web** [comment]
Having same issue still. I've logged in 4 times this week for my bazzite install using heroic launcher.
